## Домашнє завдання 6: Статистичні тести

0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [10]:
import pandas as pd
import numpy as np
import os
os.listdir ('../Data/')

['.ipynb_checkpoints',
 'ab_data.csv',
 'cookie_cats.csv',
 'data.csv',
 'data_2.csv',
 'health_insurance_cross_sell_prediction.csv',
 'mysqlsampledatabase (1).zip',
 'mysqlsampledatabase.sql',
 'supermarket_sales.csv',
 'test_db',
 'test_db-master.zip',
 'yulu_rental.csv']

In [33]:
data_path_1 = '../Data/data.csv'
df_1 = pd.read_csv(data_path_1)
df_1.head(5)

Unnamed: 0       data
0           0  17.499453
1           1  19.662399
2           2   7.182823
3           3  29.841625
4           4   9.239386

In [34]:
df_1.shape

(100, 2)

In [39]:
data_1 = df_1['data'].to_numpy()

Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [44]:
data.shape

(100,)

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

In [28]:
from statsmodels.stats.weightstats import ztest

In [37]:
mu_0 = 20

# Виконання Z-тесту
ztest_Score, p_value = ztest(data, value= mu_0, alternative='larger')

# Виведення результатів
print(f"Z-статистика: {ztest_Score}")
print(f"p-значення: {p_value}")

# Висновок
alpha = 0.05
if p_value < alpha:
    print("Відхиляємо нульову гіпотезу: середній чек статистично значущо збільшився.")
else:
    print("Не відхиляємо нульову гіпотезу: немає статистично значущого зростання середнього чеку.")

Z-статистика: -0.9953477389335053
p-значення: 0.8402164165464012
Не відхиляємо нульову гіпотезу: немає статистично значущого зростання середнього чеку.


Для виконання завдання обрано статистичний z-test, так як розмір вибірки більше 30 (таблиця data має 100 рядків, 2 колонки). Р-значення: дорівнює 0.8402164165464012, що є більше за рівень значущості (alpha = 0.05). Тому можна зробити висновок, що ми не можемо відхилити нульову гіпотезу і наразі відсутня статистично значуща різниця у розмірі середнього чеку за останній місяць. 

2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [38]:
data_path_2 = '../Data/data_2.csv'
df_2 = pd.read_csv(data_path_2)
df_2.head(5)

data
0   5.940159
1  29.189344
2  11.383502
3  15.434098
4  19.896752

In [41]:
df_2.shape

(120, 1)

In [42]:
data_2 = df_2['data'].to_numpy()

In [47]:
mean_1 = np.mean (data_1)
mean_2 = np.mean (data_2)
print ('Середній чек за липень серпень 2022 року:', mean_1)
print ('Середній чек за серпень 2022 року:', mean_2)

Середній чек за липень серпень 2022 року: 19.37755319670441
Середній чек за серпень 2022 року: 19.37322276512519


In [52]:
ztest(data_1, data_2, value=0)

(0.005246673456808504, 0.9958137794593356)

In [49]:
ztest(data_1, data_2, alternative='two-sided')

(0.005246673456808504, 0.9958137794593356)

Отже, статистика для двох вибірок z-критерію становить 0,0052, відповідне р-value дорівнює 0,9958. 
З розрахунку видно, що значення р-value >0,05, тому ми не можемо відхилити нульову гіпотезу і наразі відсутня статистично значуща різниця у розмірі середнього чеку між липнем та серпнем 2022 року. Висновок статистичного тесту збігається з спостереженням середніх значень чеків в липні та серпні 2022 року, їх значення практично однакове.